internal/sync/sync.go

```diff
@@ -126,34 +127,36 @@ func (s *Synchronizer) handleSync(apiSync bool, errChan chan error) {
 func (s *Synchronizer) sync() error {
 	go s.stateDiffCollector.Run()
 	// Get state
-	for stateDiff := range s.stateDiffCollector.GetChannel() {
+	for collectedDiff := range s.stateDiffCollector.GetChannel() {
 		start := time.Now()
 
-		err := s.updateState(stateDiff)
-		if err != nil || s.state.Root().Cmp(stateDiff.NewRoot) != 0 {
+		err := s.updateState(collectedDiff)
+		if err != nil || s.state.Root().Cmp(collectedDiff.stateDiff.NewRoot) != 0 {
 			// In case some errors exist or the new root of the trie didn't match with
 			// the root we receive from the StateDiff, we have to revert the trie
+			s.logger.With("Error", err).Error("State update failed, reverting state")
 			prometheus.IncreaseCountStarknetStateFailed()
 			s.setStateToLatestRoot()
 			return err
 		}
 		prometheus.IncreaseCountStarknetStateSuccess()
 		prometheus.UpdateStarknetSyncTime(time.Since(start).Seconds())
-		s.logger.With("Block Number", stateDiff.BlockNumber,
-			"Missing Blocks to fully Sync", int64(s.stateDiffCollector.LatestBlock().BlockNumber)-stateDiff.BlockNumber,
+		s.logger.With("Block Number", collectedDiff.stateDiff.BlockNumber,
+			"Missing Blocks to fully Sync", int64(s.stateDiffCollector.LatestBlock().BlockNumber)-
+				collectedDiff.stateDiff.BlockNumber,
 			"Timer", time.Since(start)).
 			Info("Synced block")
```

Comment:
We can avoid having `Info` on its own line and shorten the keys for better readability as follows.

```suggestion
		s.logger.With(
			"Number", collectedDiff.stateDiff.BlockNumber,
			"Pending", int64(s.stateDiffCollector.LatestBlock().BlockNumber)-collectedDiff.stateDiff.BlockNumber,
			"Time", time.Since(start),
		).Info("Synchronized block")
```

Reply:
done